src/ngraph_nnvm_ops.cc

```diff
@@ -42,33 +42,18 @@ namespace ngraph_bridge {
   { "cpu" }
 #endif
 
-void append_cached_to_forward(TensorVector *results,
-                              const std::shared_ptr<Graph> &graph,
-                              const int mode) {
-  if (results == nullptr) {
-    throw std::runtime_error(
-        "NGRAPH_BRIDGE: append_cached_to_forward recieved nullptr results");
-  }
-  results->insert(results->end(), graph->cached_aux_values[mode].begin(),
-                  graph->cached_aux_values[mode].end());
-  results->insert(results->end(), graph->cached_values[mode].begin(),
-                  graph->cached_values[mode].end());
-}
-
 void update_aux_vals(const std::shared_ptr<Graph> &graph,
+                     const TensorVector &results,
                      const std::vector<mxnet::NDArray> &inputs, const int mode,
                      const int offset = 0) {
-  const size_t cached_aux_count = graph->cached_aux_values[mode].size();
-  if (cached_aux_count > 0) {
-    std::vector<mxnet::OpReqType> aux_req;
-    std::vector<mxnet::NDArray> aux_outs;
-
-    for (size_t i = 0; i < cached_aux_count; ++i) {
-      aux_outs.push_back(inputs[graph->cached_aux_positions[mode][i] + offset]);
-      aux_req.push_back(mxnet::kWriteTo);
-    }
-
-    result_to_NDArray(graph->cached_aux_values[mode], aux_req, aux_outs, true);
+  const size_t cached_aux_count = graph->cached_aux_positions[mode].size();
+  for (size_t i = 0; i < cached_aux_count; ++i) {
+    auto buffer_size = results[i]->get_size_in_bytes();
+
+    void *mxnet_ndarray = inputs[graph->cached_aux_positions[mode][i] + offset]
+                              .storage_handle()
+                              .dptr;
+    results[i]->read(mxnet_ndarray, 0, buffer_size);
```

Comment:
is there a way to avoid this read..can we wrap corresponding tensorview in ndarray.

Reply:
No, it's also an input to the nGraph function, so we need the output to be a copy and we need to read in back into the input.